### ISSUE TITLE
NodeNetworkUnavailable condition fix not included in v3.4.0 as release notes suggest

### ISSUE THREAD

**Issue by ffilippopoulos** [projectcalico/calico#2379]
According to release notes for v3.4.0 the following fix should be included:
```
For compatibility with the GCE cloud provider, Calico now clears the Kubernetes NodeNetworkUnavailable condition. node #89 (@aarnaud)
```
should be included.
The commit for the above issue seems to only be included in `release-v3.3` branch.
https://github.com/projectcalico/node/pull/89/commits/d2f4efca880558c11bb542e84077bc9ac705fcfa

## Expected Behavior
Deploying calico-node `3.4.0` to kube-gcp cluster should mark `NodeNetworkUnavailable` as `false` on nodes that `calico-node` is running.

## Current Behavior
Had to downgrade to `quay.io/calico/node:v3.3.2` image to succeed the above

## Possible Solution
Either update the release notes or add the fix in 3.4

## Steps to Reproduce (for bugs)

Deploy calico v3.4.0 on a gcp kubernetes cluster.

## Context
Holding us from updating to calico v3.4 and use its features like bgp advertisement of services address range

## Your Environment
Kubernetes cluster running in `gcp`.

**Comment by caseydavenport** [comment]
@ffilippopoulos thanks for raising - yes, this was somehow marked for v3.4 but instead added to v3.3..

We will release a Calico v3.4.1 which addresses this. I've adjusted the Calico v3.4.0 release notes to align with reality.

**Comment by george-angel** [comment]
@caseydavenport I think we can skip v3.4.0 and go straight to v3.5.0 (released on the 26th), can you confirm v3.5.0 has this fix? Thank you.

**Comment by ffilippopoulos** [comment]
closing since both release notes are updated and v3.5.0 includes the commit
